pkg/features/client_adapter.go

```diff
@@ -0,0 +1,43 @@
+package features
+
+import (
+	"fmt"
+
+	clientfeatures "k8s.io/client-go/features"
+	"k8s.io/component-base/featuregate"
+)
+
+type clientFeatureGateAdapter struct {
+	mfg featuregate.MutableFeatureGate
+}
+
+func (a *clientFeatureGateAdapter) Enabled(name clientfeatures.Feature) bool {
+	return a.mfg.Enabled(featuregate.Feature(name))
+}
+
+func (a *clientFeatureGateAdapter) Add(in map[clientfeatures.Feature]clientfeatures.FeatureSpec) error {
```

Comment:
Would it make sense and if yes how hard it would be to write a unit test that would make sure no new field were added to `clientfeatures.FeatureSpec` ? (reflection?)

Reply:
As discussed offline, we are expecting impactful mistakes to be caught by compilation failures or panics at package-initialization time and won't include a unit test.